src/main/java/org/openrewrite/quarkus/ChangeQuarkusPropertyValue.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.quarkus;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.internal.StringUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.quarkus.search.FindQuarkusProperties;
+
+import java.util.List;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class ChangeQuarkusPropertyValue extends Recipe {
+
+    @Option(displayName = "Property key",
+            description = "The name of the property key whose value is to be changed. Supports regex.",
+            example = "quarkus.hibernate-search-orm.indexing.plan.synchronization.strategy")
+    String propertyKey;
+
+    @Option(displayName = "New value",
+            description = "The new value to be used for key specified by `propertyKey`.")
+    String newValue;
+
+    @Option(displayName = "Old value",
+            required = false,
+            description = "Only change the property value if it matches the configured `oldValue`.")
+    @Nullable
+    String oldValue;
+
+    @Option(displayName = "Profile",
+            description = "The profile where the property is defined. If not specified, the property will be changed on the default profile.",
+            required = false,
+            example = "dev")
+    @Nullable
+    String profile;
+
+    @Option(displayName = "Change for all Profiles",
+            description = "If set, thr property will be changed on all available profiles.",
```

Comment:
This probably needs a change in all recipes:
```suggestion
            description = "If set to true, the property will be changed on all available profiles.",
```